docs/docs/developers/contracts/writing_contracts/storage/notes.md

```diff
@@ -0,0 +1,78 @@
+Looking at the noir circuits in these components, you will see references to the distinction between public/private execution and state.
+
+### Lifecycle functions
+Inside the [lifecycle](
+https://github.com/AztecProtocol/aztec-packages/tree/#include_aztec_version/noir-projects/aztec-nr/aztec/src/note/lifecycle.nr) circuits we see the functions to create and destroy a note, implemented as insertions of note hashes and nullifiers respectively.
+
+We also see a function to create a note hash from the public context, a way of creating a private variable from a public call (run in the sequencer).
+
+### Note Interface functions
+To see a [note_interface](https://github.com/AztecProtocol/aztec-packages/tree/#include_aztec_version/noir-projects/aztec-nr/aztec/src/note/note_interface.nr) implementation, we will look at a simple [ValueNote](https://github.com/AztecProtocol/aztec-packages/tree/#include_aztec_version/noir-projects/aztec-nr/value-note/src/value_note.nr).
+
+#### Computing hashes and nullifiers
+A few key functions in the note interface are around computing the note hash and nullifier, with logic to get/use secret keys from the private context.
+
+In the ValueNote implementation you'll notice that it uses the `pedersen_hash` function. This is currently required by the protocol, but may be updated to another hashing function, like poseidon.
+
+As a convenience, the outer [note/utils.nr](https://github.com/AztecProtocol/aztec-packages/tree/#include_aztec_version/noir-projects/aztec-nr/aztec/src/note/utils.nr) contains an implementation of a function required by Aztec contracts using private storage: `compute_note_hash_and_nullifier`
+
+#### Serialization and deserialization
+Serialization/deserialization of content is used to convert between the Note's variables and a generic array of Field elements. The Field type is understood and used by lower level crypographic libraries.
+This is analogous to the encoding/decoding between variables and bytes in solidity.
+
+For example in ValueNote, the `serialize_content` function simply returns: the value, owner address (as a field) and the note randomness; as an array of Field elements.
+
+### Value as a sum of Note
+We recall that multiple notes are associated with a "slot" (or ID), and so the value of a numerical note (like ValueNote) is the sum of each note's value.
+The helper function in [balance_utils](https://github.com/AztecProtocol/aztec-packages/blob/#include_/noir-projects/aztec-nr/value-note/src/balance_utils.nr) implements this logic taking a `Set` of `ValueNotes`.
+
+A couple of things worth clarifying:
+- A `Set` takes a Generic type, specified here as `ValueNote`, but can be any `Note` type (for all notes in the set).
+- A `Set` of notes also specifies *the* slot of all Notes that it holds.
+
+### Abstracting Notes
```

Comment:
Would be great to get some include_code references of easy_private_state and how it looks in action